src/coinbase/address.ts

```diff
@@ -218,29 +218,33 @@ export class Address {
       createTransferRequest,
     );
 
-    const transfer = Transfer.fromModel(response.data);
-    const transaction = transfer.getTransaction();
-    let signedPayload = await this.key.signTransaction(transaction);
-    signedPayload = signedPayload.slice(2);
+    let transfer = Transfer.fromModel(response.data);
 
-    const broadcastTransferRequest = {
-      signed_payload: signedPayload,
-    };
+    if (!Coinbase.useServerSigner) {
+      const transaction = transfer.getTransaction();
+      let signedPayload = await this.key!.signTransaction(transaction);
+      signedPayload = signedPayload.slice(2);
 
-    response = await Coinbase.apiClients.transfer!.broadcastTransfer(
-      this.getWalletId(),
-      this.getId(),
-      transfer.getId(),
-      broadcastTransferRequest,
-    );
+      const broadcastTransferRequest = {
+        signed_payload: signedPayload,
+      };
 
-    const updatedTransfer = Transfer.fromModel(response.data);
+      response = await Coinbase.apiClients.transfer!.broadcastTransfer(
+        this.getWalletId(),
+        this.getId(),
+        transfer.getId(),
+        broadcastTransferRequest,
+      );
+
+      transfer = Transfer.fromModel(response.data);
+    }
 
     const startTime = Date.now();
     while (Date.now() - startTime < timeoutSeconds * 1000) {
-      const status = await updatedTransfer.getStatus();
+      await transfer.reload();
+      const status = transfer.getStatus();
```

Comment:
do we need to call `transfer.reload()` for `if(Coinbase.useServerSigner)`

Reply:
Yes because that is how we update the state of the transfer